cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzurePrivateDnsZoneRegistrationEnum.java

```diff
@@ -0,0 +1,69 @@
+package com.sequenceiq.cloudbreak.cloud.azure;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * This enum contains possible private DNS zones that are only registered in environment service, but are not used by cloudbreak.
+ */
+public enum AzurePrivateDnsZoneRegistrationEnum implements AzurePrivateDnsZoneDescriptor {
+
+    AKS("Microsoft.ContainerService/managedClusters", "managedClusters",
+            "privatelink.{region}.azmk8s.io or {subzone}.privatelink.{region}.azmk8s.io",
+            List.of(DnsZoneNamePatternConstants.AKS_DNS_ZONE_NAME_PATTERN_REGION,
+                    DnsZoneNamePatternConstants.AKS_DNS_ZONE_NAME_PATTERN_SUBZONE_AND_REGION));
+
+    private final String resourceType;
+
+    private final String subResource;
+
+    private final String dnsZoneName;
+
+    private final List<Pattern> dnsZoneNameRegexPatterns;
+
+    AzurePrivateDnsZoneRegistrationEnum(String resourceType, String subResource, String dnsZoneName, List<String> dnsZoneNameTemplate) {
+        this.resourceType = resourceType;
+        this.subResource = subResource;
+        this.dnsZoneName = dnsZoneName;
+        this.dnsZoneNameRegexPatterns = dnsZoneNameTemplate.stream().map(Pattern::compile).collect(Collectors.toList());
+    }
+
+    public String getResourceType() {
+        return resourceType;
+    }
+
+    public String getSubResource() {
+        return subResource;
+    }
+
+    @Override
+    public String getDnsZoneName() {
+        return dnsZoneName;
+    }
+
+    @Override
+    public List<Pattern> getDnsZoneNamePatterns() {
+        return dnsZoneNameRegexPatterns;
+    }
+
+    @Override
+    public String toString() {
+        return "AzurePrivateDnsZoneRegistrationEnum{" +
+                "resourceType='" + resourceType + '\'' +
+                ", subResource='" + subResource + '\'' +
+                ", dnsZoneName='" + dnsZoneName + '\'' +
+                ", dnsZoneNameRegexPatterns=" + dnsZoneNameRegexPatterns +
+                "} " + super.toString();
+    }
+
+    private static class DnsZoneNamePatternConstants {
+        private static final String AKS_DNS_ZONE_NAME_PATTERN_REGION = "privatelink\\.[a-z\\d-]+.azmk8s.io";
+
+        private static final String AKS_DNS_ZONE_NAME_PATTERN_SUBZONE_AND_REGION = "[\\w+-]+\\.privatelink\\.[a-z\\d-]+.azmk8s.io";
```

Comment:
shouldn't we make them configurable? This will avoid code changes if need to change the same in future.

Reply:
The azure private DNS zone naming should change rarely. Also, this is not a parameter of one of our environments, and the private DNS zone names are not controlled by us, and they are  constant across all of our environments. So I think they are kept better in code.

Reply:
But what if it would be changed by MS for whatever reason in the future? I would place it in config, too, it should not be a big deal

Reply:
I do not agree: this is not a config parameter of any cloudera environment, this is how we interact with a given foreign product, is uniform in all the cloudera environments, and is part of the code and tests. If you put them into yaml config files, you would have the possibility to have different parameters per cloudera environment - huge complexity for no gain in my opinion. You put things into config files that change across cloudera environments and not things that may change in the product. 

Also, please take the supportability aspect into consideration: how can you track changes to parameters in cloudera prod env, who would spot it? I guess first a customer.

If you check the code, you will find that most of the regexps are part of the code, hardcoded, and not coming from an external parameter source, I guess mostly for the same reason: part of the product, and needs to be tracked in a clean and transparent way.

Also, there are unit tests for the regex. If we change the external parameter, we will need to change the tests as well, and that already is a code change in itself. I find it cumbersome, to say the least, to unit test a configuration parameter.

Another reason not to outsource the regex strings is, that productive and test code would get even further apart.

Also, MS is expected to change it in a backward compatible mode, so these regexps should work also later on.